### PR TITLE
fix: add gitlab MR labels at time of creation

### LINKED ISSUE #985: labels not being applied to new merge requests in gitlab APIv4
I suspect the reason for this lies in this API documentation: https://docs.gitlab.com/ee/api/merge_requests.html#update-mr

The v4 API has a tendency to utilize `iid` rather than `id` for referencing merge requests.

### PR DESCRIPTION
**Pull request by rarkins**
This refactor means that createPr now takes `labels` as a new parameter so that the GitLab API can include them with the MR creation request, so save one API call. This also fixes #985 as a result.

### CODE REVIEW

**Comment by rarkins** [comment]
@Autre31415 could you take a look and let me know if you think it looks ok?

File: lib/api/github.js [hunk 1]
```diff
@@ -583,6 +585,7 @@ async function createPr(branchName, title, body, useDefaultBranch) {
     },
   })).body;
   pr.displayNumber = `Pull Request #${pr.number}`;
+  await addLabels(pr.number);
```
Comment (Autre31415) [inline lib/api/github.js]: Since this is the Github side, don't we need to pass labels here?

**Comment by Autre31415** [comment]
Would it be possible to also append labels after the fact in the case of PRs that already exist but don't have labels already?

File: lib/api/github.js (hunk 1 above)
Comment (rarkins) [reply lib/api/github.js]: yes!

**Comment by rarkins** [comment]
@Autre31415 currently we do not add labels "after the fact". The reasoning is that: (a) we only have one way to configure labels and they will be added at PR time, and (b) if someone *removes* a label we added, they surely don't want us to be adding it back again. Unless this workflow changes in future, I don't see a need for us to support adding labels to pre-existing PRs.

**Comment by rarkins** [comment]
@Autre31415 I have now removed `addLabels` from platform exports to make it clearer that ad-hoc addition of labels is not supported. `addLabels` wasn't called from anywhere else in the code anyway.

**Comment by Autre31415** [comment]
@rarkins Looks good to me. Now only the merge conflicts stand in the way.

**Comment by rarkins** [comment]
I'll apply the same changes to v10 soon

**Pull request merged**